src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,39 @@
 package core.basesyntax.service;
-
+import core.basesyntax.exception.InvalidUserException;
 import core.basesyntax.dao.StorageDao;
-import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private final StorageDao storageDao;
+
+    public RegistrationServiceImpl(StorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidUserException("User cannot be null.");
+        }
+        if (user.getLogin() == null || user.getLogin().isEmpty()) {
+            throw new InvalidUserException("Login cannot be null or empty.");
+        }
+        if (user.getLogin().length() < 6) {
+            throw new InvalidUserException("Login must be at least 6 characters long.");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new InvalidUserException("User with this login already exists.");
+        }
```

Comment:
The check `storageDao.get(user.getLogin()) != null` ensures that no duplicate users are registered. Ensure that the `StorageDao` implementation correctly handles this check.